tuntap-unix-darwin.c

```diff
@@ -36,24 +36,74 @@
 #include "tuntap.h"
 #include "private.h"
 
+
+#include <sys/kern_control.h>
+#include <sys/sys_domain.h>
+#include <sys/kern_event.h>
+
+#define APPLE_UTUN "com.apple.net.utun_control"
+#define UTUN_OPT_IFNAME 2
+
+static int
+tuntap_utun_open(char * ifname, uint32_t namesz)
+{
+	int fd;
+
+	fd = socket(PF_SYSTEM, SOCK_DGRAM, SYSPROTO_CONTROL);
+	if(fd == -1)
+		return -1;
+
+	struct ctl_info info;
```

Comment:
Variable declarations should respect ANSI C and thus be at the top of the current block.